var/spack/repos/builtin/packages/py-parsl/package.py

```diff
@@ -16,16 +16,21 @@ class PyParsl(PythonPackage):
 
     maintainers("hategan")
 
+    version(
+        "2023.08.21", sha256="d7d6145ad5ab63baf9c9f9441a0a6ea5be6f896ef8094d47bf64d949a56b1782"
+    )
     version("1.2.0", sha256="342c74ee39fa210d74b8adfb455f0a9c20d9f059ec5bd9d60c5bdc9929abcdcc")
     version("1.1.0", sha256="6a623d3550329f028775950d23a2cafcb0f82b199f15940180410604aa5d102c")
 
     variant("monitoring", default=False, description="enable live monitoring")
     # See https://parsl.readthedocs.io/en/stable/userguide/monitoring.html
 
-    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("python@3.8:", type=("build", "run"), when="@2023.08.21:")
+    depends_on("python@3.6:", type=("build", "run"), when="@:1.2")
     depends_on("py-setuptools", type="build")
     depends_on("py-pyzmq@17.1.2:", type=("build", "run"))
-    depends_on("py-typeguard@2.10:", type=("build", "run"))
+    depends_on("py-typeguard@2.10:2", type=("build", "run"), when="@2023.08.21:")
+    depends_on("py-typeguard@2.10:", type=("build", "run"), when="@:1.2")
     depends_on("py-typing-extensions", type=("build", "run"))
```

Comment:
Has version reqs in the latest version

Reply:
Missing six dep

Reply:
not sure how i missed these, but thanks for checking